docs/architecture/adr-071-proposer-based-timestamps.md

```diff
@@ -0,0 +1,458 @@
+
+ * Proposed
+
+## Context
+
+Tendermint currently provides a monotonically increasing source of time known as [BFTTime](https://github.com/tendermint/spec/blob/master/spec/consensus/bft-time.md).
+This mechanism for producing a source of time is reasonably simple.
+Each correct validator adds a timestamp to each `Precommit` message it sends.
+The timestamp it sends is either the validator's local time or one millisecond greater than the previous block time, depending on which value is greater.
+When a block is produced, its block timestamp is chosen as the weighted median of the times in all of the `Precommit` message.
+The weighting is proportional to the amount of voting power, or stake, a validator has on the network.
+This mechanism for producing timestamps is both deterministic and byzantine fault tolerant.
+ 
+This current mechanism for producing timestamps has a few drawbacks.
+Validators do not have to agree at all on how close the selected block timestamp is to their own local time.
+Additionally, any amount of voting power `>1/3` may directly control the block timestamp.
+As a result, it is quite possible that the timestamp is not particularly meaningful.
+
+These drawbacks present issues in the Tendermint protocol.
+Timestamps are used by light clients to verify blocks.
+Light clients rely on correspondence between their local time and the block timestamp to verify blocks they see; 
+However, their local time may be greatly divergent from the block timestamp as a result of the limitations of `BFTtime`.
+
+The proposer-based timestamps specification suggests an alternative approach for producing block timestamps that remedies these issues.
+Proposer-based timestamps alter the current mechanism for producing block timestamps in two main ways:
+
+1. The block proposer is amended to offer up its local time as the timestamp for the next block. 
+1. Correct validators only approve the proposed block timestamp if it is close enough to their own local view of time. 
+
+The result of these changes is a more meaningful timestamp that cannot be controlled by `<= 2/3` of the validator voting power. 
+This document outlines the necessary code changes in Tendermint to implement the corresponding [proposer-based timestamps specification](https://github.com/tendermint/spec/tree/master/spec/consensus/proposer-based-timestamp).
+
+## Alternative Approaches
+
+### Remove timestamps altogether
+
+Computer clocks are bound to skew for a variety of reasons.
+Using timestamps in our protocol means either accepting the timestamps as not reliable or impacting the protocol’s liveness guarantees.
+This design requires impacting the protocol’s liveness in order to make the timestamps more reliable.
+An alternate approach is to remove timestamps altogether from the block protocol.
+`BFTTime` is deterministic but may be arbitrarily inaccurate.
+This approach was not selected because a reliable source of time is quite useful for blockchain applications.
```

Comment:
> "Time" as a signifier of order within the consensus protocol... 
> For this design, I think (1) is the important axis: We already couldn't distinguish between ambient clock skew and a node intentionally choosing a "wrong" time for some reason.

We effectively already have a way to determine the order of the records in the consensus protocol, the block height. The current `BFTTime` implementation also _does_ gives us an ordering of blocks already because it is monotonically increasing. Since we can already derive an ordering given the current block data I don't think it would be worth making this change _solely_ on the understanding that it produces an order of blocks. Based on the other comments in the doc, it seems possible I did not clarify strongly enough why it's important to include a more accurate view of time from a blockchain protocol and application perspective. As Josef mentioned, light client and IBC depend on time for verification. They check that their local time is within a bound of the block they are verifying. I can add some additional information on this for clarity into the document.

Reply:
@creachadair I added some additional uses for a meaningful source of time in blockchain protocols and applications. I'm going to resolve this comment thread for now. Please re-open if this seems insufficient.